File: solutions/go/0242-valid-anagram.go
package main

/* func isAnagram2(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}

	for _, char := range s {
		if index := strings.Index(t, string(char)); index > -1 {
			t = t[:index] + t[index+1:]
			continue
		}
		return false
	}

	return true
} */

func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}

	counts := make(map[rune]int, len(s))

	for _, char := range s {
		counts[char]++
	}

	for _, char := range t {
		if counts[char]--; counts[char] < 0 {
			return false
		}
	}

	return true
}

File: solutions/php/0242-valid-anagram.php
<?php

class Solution
{
    /**
     * @param String $s
     * @param String $t
     * @return Boolean
     */
    /* function isAnagram(string $s, string $t): bool
    {
        if (strlen($s) != strlen($t)) {
            return false;
        }

        $counts = count_chars($s, 1);

        foreach (str_split($t) as $char) {
            $index = ord($char);

            if (!isset($counts[$index])) {
                return false;
            }

            if (--$counts[$index] < 0) {
                return false;
            }
        }

        return true;
    } */

    function isAnagram(string $s, string $t): bool
    {
        return count_chars($s) === count_chars($t);
    }
}
